app/components/TodoDetails.tsx

```diff
@@ -0,0 +1,135 @@
+"use client";
+import { TodoCardInfo } from "@/types";
+import {
+  PencilSquareIcon,
+  ClipboardDocumentCheckIcon,
+} from "@heroicons/react/16/solid";
+import { useState } from "react";
+import { useForm } from "react-hook-form";
+
+export default function TodoDetails({
+  information,
+  onEdit,
+}: {
+  information: TodoCardInfo;
+  onEdit: (updatedTodo: TodoCardInfo) => void;
+}) {
+  const [isEditing, setIsEditing] = useState(false);
+
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<TodoCardInfo>({
+    defaultValues: {
+      title: information.title,
+      description: information.description,
+      status: information.status,
+      type: information.type,
+    },
+  });
+
+  const onSubmit = (editedInformation: TodoCardInfo) => {
+    onEdit(editedInformation);
```

Comment:
check if editedInformation isnt undefined or invalid if (editedInformation)

Reply:
I am using react-hook-form and validation is described at the input field itself or I am not understanding something?

Reply:
Yes you are correct, my bad